app/deed/server.py

```diff
@@ -70,13 +71,20 @@ def delete(id_):
         if deed is None:
             abort(status.HTTP_404_NOT_FOUND)
         else:
-            return jsonify(id=id_), status.HTTP_200_OK
+            return {'id': id_}, status.HTTP_200_OK
 
     @blueprint.route('/deed/<deed_id>/<borrower_id>/signature/',
                      methods=['POST'])
     def sign(deed_id, borrower_id):
+        def find_deed():
+            deed_ = Deed.get(deed_id)
+            if deed_ is None:
+                abort(status.HTTP_404_NOT_FOUND)
+            return deed_
+
         def sign_allowed():
-            return Deed.matches(deed_id, borrower_id)
+            return Deed.matches(deed_id, borrower_id) and not \
```

Comment:
not sure why the response is being inverted here as it strikes me as a bit odd  but if there is a good reason for it thats fine